rust/worker/src/segment/record_segment.rs

```diff
@@ -810,27 +815,25 @@ impl RecordSegmentReader<'_> {
         Ok(data)
     }
 
-    /// Returns all ids in the record segment sorted.
-    pub(crate) async fn get_all_user_ids(&self) -> Result<Vec<&str>, Box<dyn ChromaError>> {
-        let mut ids = Vec::new();
-        let max_size = self.user_id_to_id.count().await?;
-        for i in 0..max_size {
-            let res = self.user_id_to_id.get_at_index(i).await;
-            match res {
-                Ok((_, user_id, _)) => {
-                    ids.push(user_id);
-                }
+    /// Returns all offset_ids in the record segment sorted.
+    pub(crate) async fn get_all_offset_ids(&self) -> Result<Vec<u32>, Box<dyn ChromaError>> {
+        let offset_id_count = self.count().await?;
+        join_all((0..offset_id_count).map(|i| async move {
+            match self.id_to_user_id.get_at_index(i).await {
+                Ok((_, offset_id, _)) => Ok(offset_id),
                 Err(e) => {
                     tracing::error!(
-                        "[GetAllData] Error getting user id for index {:?}: {:?}",
+                        "[GetAllData] Error getting offset id for index {}: {}",
                         i,
                         e
                     );
-                    return Err(e);
+                    Err(e)
                 }
             }
-        }
-        Ok(ids)
+        }))
+        .await
+        .into_iter()
+        .collect()
```

Comment:
Is there a reason we changed this to be a iterator with join_all + map? My concern is that into_iter + collect may result in an additional traversal over the data. I am not sure if it does though.

Reply:
Refactored with `try_collect`